### ISSUE TITLE
Add `explain` and `performance` modes

### ISSUE THREAD

**Issue by pplonski** [mljar/mljar-supervised#100]
There should be `explain` mode in the AutoML which will produce explanations.

There should be `performance` mode for max accuracy of models from AutoML.

**Comment by pplonski** [comment]
I set 3 modes:
- `Explain`
- `Perform`
- `Compete`

## Explain

It is aimed to be used when the user wants to explain and understand the data.
It is using 75%/25% train/test split. It is using: `Baseline`, `Linear`, `Decision Tree`, `Random Forest`, `Xgboost` and `Neural Network` algorithms. It has full explanations: learning curves, importance plots, and SHAP plots

## Perform

It should be used when the user wants to train a model that will be used in real-life use cases.
It is using 5-fold CV.  It is using: `Linear`, `Random Forest`, `LightGBM`, `Xgboost`, `CatBoost` and `Neural Network`. It uses ensembling. It has learning curves and importance plots in reports.

## Compete

It should be used for competitions.
It is using 10-fold CV. It is using: `Linear`, `Decision Tree`, `Random Forest`, `Extra Trees`, `LightGBM`, `Xgboost`, `CatBoost`, `Neural Network` and `Nearest Neighbors`. It has only learning curves in the reports.
